packages/cli/src/commands/init/init.ts

```diff
@@ -268,6 +271,55 @@ function createTemplateUri(options: Options, version: string): string {
   return options.template || `react-native@${version}`;
 }
 
+/*
+Starting from 0.73, react-native.config.js is created by CLI during the init process.
+It contains automaticPodsInstallation flag set to true by default.
+This flag is used by CLI to determine whether to install CocoaPods dependencies when running ios commands or not.
+It's created by CLI rather than being a part of a template to avoid displaying this file in the Upgrade Helper,
+as it might bring confusion for existing projects where this change might not be applicable.
+For more details, see https://github.com/react-native-community/cli/blob/main/docs/projects.md#projectiosautomaticpodsinstallation
+*/
+function createDefaultConfigFile(directory: string) {
+  const content = `module.exports = {
+  project: {
+    ios: {
+      automaticPodsInstallation: true,
+    },
+  },
+};
+`;
+
+  const filepath = 'react-native.config.js';
+  try {
+    if (!doesDirectoryExist(path.join(directory, filepath))) {
+      fs.createFileSync(filepath);
+      fs.writeFileSync(filepath, content, {encoding: 'utf-8'});
+    } else {
+      const tempFilePath = path.join(directory, 'tempConfig.js');
+      fs.writeFileSync(tempFilePath, content);
+      const cliConfigFile = require(tempFilePath);
+      const existingConfigFile = require(path.join(directory, filepath));
```

Comment:
Seems inefficient to write the content to disk, then read it back just to merge with existing config. Why can't we just use the in-memory config directly?